### DISCUSSION TITLE
How to enforce application-specific constraints?

### DISCUSSION THREAD

**Discussion by pablo-de-andres** [simphony/simphony-osp#731] [Ideas]
In GitLab by @yoavnash on Jan 9, 2020, 16:42

Two alternatives:
1. Modules/multiple ontologies
1. Data models

A *module* is an ontology fragment in which context-specific constraints are enforced by subclassing of classes in common ontology classes. 

A *data model* is written in a separate file in YAML format. Its main use is to enforce context-specific constraints whenever data is pushed is to the dataspace. It is not allowed to create new classes in a data model file, only to strengthen existing constraints or provide new ones.

**Comment by pablo-de-andres** [comment]
In GitLab by @yoavnash on Jan 9, 2020, 16:48

changed the description

**Comment by pablo-de-andres** [comment]
In GitLab by @yoavnash on Jan 9, 2020, 16:51

Advantages of modules:
- An approach that utilize existing mechanisms.

Advantages of data models:
- Reduces complexity by avoiding the need for subclassing.

**Comment by pablo-de-andres** [comment]
In GitLab by @urbanmatthias on Mar 11, 2020, 08:21

I think it should be possible to add constraints to already defined entities using owl ontologies / modules.

**Comment by urbanmatthias** [comment]
Another idea: https://www.w3.org/TR/shacl/
